show.md

```diff
@@ -0,0 +1,207 @@
+
+### Implicit `return`
+
+When defining functions, we could allow eliding the `return` keyword in the return expression (final line) of the function body. This follows Rust, but note that Rust uses semicolons and KCL does not, so there is a different feel.
+
+This would make functions a bit more like the top-level of a module and I think that would be intuitive. It would also match the usual semantics for blocks in programming languages (and how I imagine they would work in KCL). However, one difference is that we should require explicit `return` on other lines of a function body, whereas at the top-level this is not needed.
+
+Not requiring explicit return elsewhere in the function would make it a bit easy to have weird bugs, although this could be addressed somewhat with syntax highlighting in the editor. It would also have different semantics to the top-level where we effectively return all values and continue executing, whereas in functions we return one value and terminate execution (we could allow multiple implicit returns as sugar for returning a tuple of value and explicit returns for early returns, but I think that is getting a bit weird).
+
+## Open questions
+
+### Sending construction geometry to the engine
+
+In theory, KCL should only send rendered geometry to the engine and construction geometry should only be used locally. However, that doesn't fit the current design. For example, we might create a line as construction geometry and then terminate a rendered line at the centre of the line. The engine would require knowledge of the construction line. We could send all geometry to the engine, labelled as construction/rendered or we could try to infer which geometry is required to be sent to the engine. I would hope the latter is purely an optimisation of the former, but haven't tried to prove that out.
+
+### Construction geometry within sketches
+
+E.g, (from [#1553](https://github.com/KittyCAD/modeling-app/issues/1553)):
+
+```
+startSketchOn('XY')
+  |> startProfileAt()
+  |> line()
+  |> line()
+  |> construction(circle())
+  |> line()
+  |> construction(line({from:[], to:[]}))
+```
+
+Neither implicit nor explicit `show` address that very well. We could support a `construction` function specifically for use within sketches/pipelines. However, I think that fundamentally, construction geometry does not fit well with the ordering ethos of pipelining. I've been thinking of a block syntax for sketching which does not have an ordering, and in that case, construction geometry works similarly to functions. For example, the above example would become:
+
+```
+sketch(on = XY) {
+  l1 = line(...)
+  l2 = line(...)
+  cnstCircle = circle(...)
+  cnstLine(...)
+  l3 = line(...)
+  enclose(l1, l2, l3)
+}
+```
```

Comment:
I have only ever used Construction Geometry with regards to sketches, so I'll reserve my comments for only sketches. I don't believe I've ever seen construction geometry for 3D objects, but it may just be a tool I've never used.

The words "Construction Geometry" actually might be causing some miscommunication. I'm not positive. For sketches that use the construction (not rendered) lines, I'll use "References" (used in NX).

Probably no surprise, but this code makes the most sense to me. The distribution of rendered entities vs reference entities is (guessing) 90/10. I'd rather the user be explicit about what needs to not be rendered, and having a `reference`, `construction`, etc. is best aligned with their workflow